<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
 <head>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <title>Manejo de errores</title>

 </head>
 <body><div class="manualnavbar" style="text-align: center;">
 <div class="prev" style="text-align: left; float: left;"><a href="sca.examples.structures.html">Trabajar con estructuras de datos</a></div>
 <div class="next" style="text-align: right; float: right;"><a href="ref.sca.html">Funciones de SCA</a></div>
 <div class="up"><a href="sca.examples.html">Ejemplos</a></div>
 <div class="home"><a href="index.html">PHP Manual</a></div>
</div><hr /><div id="sca.examples.errorhandling" class="section">
  <h2 class="title">Manejo de errores</h2>
  <p class="para"> 
   Esta sección describe cómo se manejan los errores. Existen dos
   tipos de errores:
  </p>
  <ul class="itemizedlist">
   <li class="listitem">
    <p class="para"> 
     Las excepciones en tiempo de ejecución de SCA son aquellas que señalan
     problemas en el manejo de la ejecución de componentes, y en la
     interacción con servicios remotos. Podrían ocurrir debido a
     problemas de red o de configuración.
    </p>
   </li>
   
   <li class="listitem">
    <p class="para"> 
     Las excpeciones de negocio son aquellas definidas por el
     programador. Amplían la clase Exception de PHP, y son lanzadas
     y capturadas deliberadamente como parte de la lógica de negocio.
    </p>
   </li>
   
  </ul>
  
  <div class="section" id="sca.examples.errorhandling.runtime">
   <h2 class="title">Manejo de excepciones en tiempo de ejecución</h2>
   <p class="para"> 
    Hay dos tipos de excepciones en tiempo de ejecución de SCA: 
   </p>
   
   <ul class="itemizedlist">
    <li class="listitem">
     <p class="para"> 
      SCA_RuntimeException - señala un problema encontrado por o
      quizás ocurrido dentro del tiempo de ejecución de SCA. Puede ser lanzada por
      varias razones, muchas de las cuales pueden ocurrir sin tener en cuenta si
      si se ha realizado una conexión a un servicio local o remoto:
      un error en una de las anotaciones de un componente, un fichero
      WSDL o php faltante, etc. En el caso de servicios web, una
      SCA_RuntimeException también puede ser lanzada si se recibe un
      SoapFault desde un servicio web remoto y el código de error del
      SoapFault indica que sea improbable que tenga éxito un reintento.
     </p>
    </li>
    
    <li class="listitem">
     <p class="para"> 
      SCA_ServiceUnavailableException - esta es una subclase de
      SCA_RuntimeException y señala un problema al conectarse a, o
      usar un servicio remoto, aunque podría tener éxito si se reintenta.
      En el caso de servicios web, esta excepción es lanzada si se
      recibe un SoapFault con un código de error que indique que un
      reintento podría tener éxito.
     </p>
    </li>
    
   </ul>
  </div>
  
  <div class="section" id="sca.examples.errorhandlilng.business"> 
   <h2 class="title">Manejo de excepciones de negocio</h2>
   <p class="para"> 
    Las excepciones de negocio pueden ser definidas y lanzadas por un componente
    de la manera usual, sin tener en cuenta si el componente ha sido
    invocado local o remotamente. El tiempo de ejecución de SCA no captura
    las excepciones de negocio que han sido lanzadas por un componente invocado
    localmente, por lo que serán devueltas al llamador de la manera usuarl. Por otro
    lado, si un componente ha sido invocado mediante un servicio web, el tiempo
    de ejecución de SCA, al final de proporcionar el servicio, captura las excepciones
    de negocio, y se asegurará de que sean pasadas otra vez al final
    de la llamada y sena relanzadas. Asumiento que al final de la llamada hay una definición
    de la excepción (esto es, es capaz de incluir un fichero que contenga la
    clase de PHP que contenga la excepción) la excepción relanzada
    contendrá los mismos detalles que la original, por lo que los métodos
    <span class="function"><strong>getLine()</strong></span> y
    <span class="function"><strong>getFile()</strong></span>, por ejemplo, contendrán
    la ubicación de donde fue lanzada la excepción dentro de la lógica
    de negocio. La excepción será pasada en el campo &#039;detail&#039; de un error
    soap con el código de error &quot;Client&quot;. 
   </p>
  </div>
  
  
 </div><hr /><div class="manualnavbar" style="text-align: center;">
 <div class="prev" style="text-align: left; float: left;"><a href="sca.examples.structures.html">Trabajar con estructuras de datos</a></div>
 <div class="next" style="text-align: right; float: right;"><a href="ref.sca.html">Funciones de SCA</a></div>
 <div class="up"><a href="sca.examples.html">Ejemplos</a></div>
 <div class="home"><a href="index.html">PHP Manual</a></div>
</div></body></html>
